benchto-service/src/main/java/io/trino/benchto/service/BenchmarkService.java

```diff
@@ -159,6 +170,31 @@ public void finishExecution(String uniqueName, String benchmarkSequenceId, Strin
         LOG.debug("Finishing execution - {}", execution);
     }
 
+    private List<Measurement> normalizeMeasurements(List<FlatMeasurement> input)
```

Comment:
It looks like `READ_COMMITED` isolation can be not sufficient here, wdyt?
We snaphost the table (`findAll`) and after all we make some chnages basing on that snaphost (possibly obsolete).

Reply:
Great catch, I forgot benchmarks can be run concurrently. Any ideas on how to best handle it?

Reply:
I think you should use here `SERIALIZABLE` isolation level.

Reply:
Maybe it would be easier to use an explicit lock for the `metrics` table? Processing metrics here should be relatively fast and is done at the end of a benchmark. Retrying the whole transaction is more complex.

Reply:
Yes, it is a good idea.

Reply:
Done!